tests/ui/fmt/ptr-metadata.rs

```diff
@@ -0,0 +1,19 @@
+//@ run-pass
+//@ check-run-results
+//@ normalize-stdout-test: "0x[0-9a-fA-F]+" -> "$$HEX"
+
+use std::fmt::Display;
+
+fn main() {
```

Comment:
can you make this a core unit test instead?

Reply:
Sure no problem, I'll fix whatever my reviewer(s) ask me to do.

Would you mind elaborating a bit on why though? Do you prefer it as a core  unit test because then the test can be run with a stage0 compiler, maybe? What I like about having it as a UI test is that `--bless` works when the output change.

Reply:
Done. I had to (with the current test code) add "regex" as an allowed library dependency, but it is already present as an allowed rustc dependency, so it's fine I think.